src/cascadia/WindowsTerminal/WindowsTerminal.rc

```diff
@@ -53,6 +53,16 @@ END
 // Icon with lowest ID value placed first to ensure application icon
 // remains consistent on all systems.
 IDI_APPICON             ICON                    "..\\..\\..\\res\\terminal.ico"

```

Comment:
this feels like the sort of thing that would be appropriate to use #ifdefs on -- there's no need to ship all icons in all variants, and no need to spend CPU at runtime to compute the build variant (imho).  does your project system expose any preprocessor macros for this?

i'm suggesting something like

```
#if OFFICIAL_BUILD
IDI_APPICON ICON "res\\terminal.ico"
IDI_APPICON_HC ICON "res\\terminal_contrast-black.ico"
#elif PREVIEW_BUILD
IDI_APPICON ICON "res\\terminal-Pre.ico"
IDI_APPICON_HC ICON "res\\terminal-Pre_contrast-black.ico"
#else
IDI_APPICON ICON "res\\terminal-Dev.ico"
IDI_APPICON_HC ICON "res\\terminal-Dev_contrast-black.ico"
#endif
```

Reply:
So, right now Terminal _doesn't build different variants_ for its different packages.

Our original intent with this build tooling was that a single build would produce both packages from the same assets such that our ring promotion engineer (so, me) could take the same package and stage it for both Preview and Stable.

Given that we re-spin every stable build anew off backported changes from the last preview, I'm not sure we're benefitting from that, but I'm not sure this is the place to change that. I'll happily hold this change back until I do that other thing.